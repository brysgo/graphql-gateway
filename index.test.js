const { parse, buildASTSchema } = require("graphql");
const graphqlGateway = require("./");

const schema1 = buildASTSchema(
  parse(`
    type Query {
      this(hasAnArgument: Int): [Bar]!
      thing: Int!
    }
    type Bar {
      id: ID!
      a: Int
    }
    `)
);

const schema2 = buildASTSchema(
  parse(`
    type Query {
      this(hasADifferentArgument: String): [Bar]!
      random: String
    }
    type Bar {
      id: ID!
      is: String
    }
    `)
);

test("splitting requests and merging results of two simple schemas", function() {
  const metaSchema = [
    {
      schema: schema1,
      resolve: function(query) {
        expect(query).toMatchSnapshot();
        return {
          this: [{ id: "foo", a: 2 }, { id: "bar", a: 3 }],
          thing: 32
        };
      }
    },
    {
      schema: schema2,
      resolve: function(query) {
        expect(query).toMatchSnapshot();
        return {
          this: [{ id: "foo", is: "two" }, { id: "bar", is: "three" }],
          random: "stuff"
        };
      }
    }
  ];
  const result = graphqlGateway(
    metaSchema,
    `
    query SomeQuery {
      this(hasAnArgument: 5, hasADifferentArgument: "foo") {
        id
        is
        a
      }
      random
      thing
    }
  `
  );
  expect(result).toMatchSnapshot();
});
