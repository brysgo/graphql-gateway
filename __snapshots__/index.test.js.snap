// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitting requests and merging results of two simple schemas 1`] = `
"query SomeQuery {
  this(hasAnArgument: 5) {
    id
    a
  }
  thing
}
"
`;

exports[`splitting requests and merging results of two simple schemas 2`] = `
"query SomeQuery {
  this(hasADifferentArgument: \\"foo\\") {
    id
    is
  }
  random
}
"
`;

exports[`splitting requests and merging results of two simple schemas 3`] = `
Object {
  "random": "stuff",
  "thing": 32,
  "this": Array [
    Object {
      "a": 2,
      "id": "foo",
      "is": "two",
    },
    Object {
      "a": 3,
      "id": "bar",
      "is": "three",
    },
  ],
}
`;
